### PR TITLE
[TSK-286] add Confirm page

### PR DESCRIPTION
**Pull request by Hoocki**

### CODE REVIEW

File: frontend/src/pages/main/payment/components/confirm/Confirm.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        if (!receivedUserWallet) return;
+        const payment = createPayment(loggedUser.email,
+                                                                receivedUserWallet.id,
+                                                                merchant.email,
+                                                                receivedUserWallet.id,
+                                                                qrCodeData.money,
+                                                                isChecked);
+        await makePayment(payment);
+        dispatch(showNotification({show: true, type: NotificationType.SUCCESS, message: Notifications.SUCCESSFUL_PAYMENT}));
+        navigate(Paths.ANY);
```
Comment (adsukharev) [inline frontend/src/pages/main/payment/components/confirm/Confirm.tsx]: maybe home?

File: frontend/src/pages/main/payment/components/confirm/Confirm.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    const dispatch = useAppDispatch();
+
+    const handleCancel = () => {
+        navigate(Paths.ANY);
+    }
+
+    const handleConfirm = async () => {
+        const receivedMerchantWallet = await getWalletById(qrCodeData.walletId);
+        if (!receivedMerchantWallet) return;
+        const merchant = await getUserService(receivedMerchantWallet.userId, UserType.Merchant);
```
Comment (adsukharev) [inline frontend/src/pages/main/payment/components/confirm/Confirm.tsx]: too many actions. Store all importnant data about merchant in qr code

File: frontend/src/pages/main/payment/components/confirm/Confirm.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        if (!merchant) return;
+        const receivedUserWallet = await getWalletByUserService(loggedUser.id, loggedUser.userType);
+        if (!receivedUserWallet) return;
+        const payment = createPayment(loggedUser.email,
```
Comment (adsukharev) [inline frontend/src/pages/main/payment/components/confirm/Confirm.tsx]: do you find this name weird ? I expect here you make actual payment. but this is just mapper.

File: frontend/src/pages/main/payment/components/confirm/Confirm.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                                                                receivedUserWallet.id,
+                                                                merchant.email,
+                                                                receivedUserWallet.id,
```
Comment (adsukharev) [inline frontend/src/pages/main/payment/components/confirm/Confirm.tsx]: why do you pass 2 times  receivedUserWallet.id ?

File: frontend/src/pages/main/payment/components/confirm/Confirm.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    const [isChecked, setIsChecked] = useState<boolean>(false);
+    const loggedUser = useAppSelector(selectLoggedUser);
+    const dispatch = useAppDispatch();
+
+    const handleCancel = () => {
+        navigate(Paths.ANY);
+    }
+
+    const handleConfirm = async () => {
+        const receivedMerchantWallet = await getWalletById(qrCodeData.walletId);
```
Comment (adsukharev) [inline frontend/src/pages/main/payment/components/confirm/Confirm.tsx]: Get it beforehand, during component creation

File: frontend/src/pages/main/payment/components/confirm/Confirm.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        if (!receivedMerchantWallet) return;
+        const merchant = await getUserService(receivedMerchantWallet.userId, UserType.Merchant);
+        if (!merchant) return;
+        const receivedUserWallet = await getWalletByUserService(loggedUser.id, loggedUser.userType);
```
Comment (adsukharev) [inline frontend/src/pages/main/payment/components/confirm/Confirm.tsx]: maybe merhcant wallet?

File: frontend/src/pages/main/payment/components/confirm/Confirm.tsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+import {getUserService} from "../../../../../services/user";
+import {UserType} from "../../../../../types/user";
+import {createPayment} from "../../../../../common/utils/mappers";
+import {useAppDispatch, useAppSelector} from "../../../../../stores/redux/hooks";
+import {selectLoggedUser} from "../../../../../stores/redux/loggedUser/loggedUserSlice";
+import {showNotification} from "../../../../../stores/redux/notification/notificationSlice";
+import {NotificationType} from "../../../../../types/notification";
+import {Notifications} from "../../../../../common/constansts/notifications";
+
+const Confirm: React.FC<ConfirmProps> = ({qrCodeData}) => {
```
Comment (adsukharev) [inline frontend/src/pages/main/payment/components/confirm/Confirm.tsx]:
```suggestion
const PaymentConfirmation: React.FC<ConfirmProps> = ({qrCodeData}) => {
```

File: frontend/src/pages/main/payment/components/confirm/Confirm.tsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+    const navigate = useNavigate();
+    const [isChecked, setIsChecked] = useState<boolean>(false);
```
Comment (adsukharev) [inline frontend/src/pages/main/payment/components/confirm/Confirm.tsx]: specify name

File: frontend/src/pages/main/payment/components/confirm/Confirm.tsx [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    const loggedUser = useAppSelector(selectLoggedUser);
+    const dispatch = useAppDispatch();
+
+    const handleCancel = () => {
+        navigate(Paths.ANY);
```
Comment (adsukharev) [inline frontend/src/pages/main/payment/components/confirm/Confirm.tsx]: to home page

File: frontend/src/pages/main/payment/components/confirm/amount/AmountField.tsx [hunk 10]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import {AmountFieldProps} from "./props";
+import {TextField} from "@mui/material";
+import {InputHelperMessage} from "../../../../../../common/constansts/inputHelperMessage";
+import {DISABLE_FIELD} from "../../../../../../common/constansts/fields";
+
+const AmountField: React.FC<AmountFieldProps> = ({amount}) => {
```
Comment (adsukharev) [inline frontend/src/pages/main/payment/components/confirm/amount/AmountField.tsx]: why did you add it here and not to common?

File: frontend/src/pages/main/payment/components/confirm/Confirm.tsx (hunk 2 above)
Comment (adsukharev) [reply frontend/src/pages/main/payment/components/confirm/Confirm.tsx]: if you need to make backend changes, add it to the list in payment automation task

File: frontend/src/pages/main/payment/components/confirm/Confirm.tsx (hunk 1 above)
Comment (Hoocki) [reply frontend/src/pages/main/payment/components/confirm/Confirm.tsx]: with home, it's not redirect to `home `(idk why), so i put `any`

File: frontend/src/pages/main/payment/components/confirm/amount/AmountField.tsx (hunk 10 above)
Comment (Hoocki) [reply frontend/src/pages/main/payment/components/confirm/amount/AmountField.tsx]: it already exists component `Amount` (i didn't use it, because it has logic that conflict with Confirm page), so i decided to put `AmountField` in Confirm components

Comment (adsukharev) [reply frontend/src/pages/main/payment/components/confirm/amount/AmountField.tsx]: in that case name it differently

**Review by adsukharev — APPROVED** [review]
I don't have time today to review it properly. To not stop your progress, I review it again in your next PR

File: frontend/src/components/fields/amountGenerator/AmountFieldGenerator.tsx [hunk 11]
```diff
@@ -1,10 +1,10 @@
 import React from 'react';
-import {AmountProps} from "./props";
+import {AmountFieldGeneratorProps} from "./props";
 import {TextField} from "@mui/material";
 import {InputHelperMessage} from "../../../common/constansts/inputHelperMessage";
 import {LOWER_AMOUNT_LIMIT, UPPER_AMOUNT_LIMIT} from "../../../common/constansts/validation";
 
-const Amount: React.FC<AmountProps> = ({updateAmountField, updateIsValidAmountField}) => {
+const AmountFieldGenerator: React.FC<AmountFieldGeneratorProps> = ({updateAmountField, updateIsValidAmountField}) => {
```
Comment (adsukharev) [inline frontend/src/components/fields/amountGenerator/AmountFieldGenerator.tsx]: very bad name, you are not generating here anything

File: frontend/src/pages/main/payment/components/confirm/PaymentConfirmation.tsx [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    const getUserWallet = async () => {
+        const receivedWallet = await getWalletByUserService(loggedUser.id, loggedUser.userType);
+        if (!receivedWallet) return;
+        setUserWallet(receivedWallet);
+    }
+
+    const handleConfirm = async () => {
+        const paymentData = createPaymentData(
```
Comment (adsukharev) [inline frontend/src/pages/main/payment/components/confirm/PaymentConfirmation.tsx]: map

Comment (Hoocki) [reply frontend/src/pages/main/payment/components/confirm/PaymentConfirmation.tsx]: don't understand what you mean

Comment (adsukharev) [reply frontend/src/pages/main/payment/components/confirm/PaymentConfirmation.tsx]: it's a mapper method. 
```suggestion
        const paymentData = mapPaymentData(
```
